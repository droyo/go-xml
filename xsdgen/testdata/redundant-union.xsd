<?xml version="1.0" encoding="UTF-8"?>
<!-- edited with XMLSpy v2008 (http://www.altova.com) by A (A) -->
<xs:schema xmlns:xs="http://www.w3.org/2001/XMLSchema"
    xmlns="http://example.org/" targetNamespace="http://example.org/">
    <xs:element name="Type1">
        <xs:complexType>
            <xs:sequence minOccurs="1" maxOccurs="1">
                <xs:element name="Elem" type="redundantUnionType" minOccurs="0" maxOccurs="unbounded"/>
            </xs:sequence>
        </xs:complexType>
    </xs:element>

    <xs:simpleType name="enumType">
        <xs:restriction base="xs:string">
            <xs:enumeration value="A"/>
            <xs:enumeration value="B"/>
            <xs:enumeration value="C"/>
        </xs:restriction>
    </xs:simpleType>

    <xs:simpleType name="redundantUnionType">
        <xs:union memberTypes="enumType xs:string"/>
    </xs:simpleType>

</xs:schema>
