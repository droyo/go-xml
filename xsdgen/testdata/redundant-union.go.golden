// Code generated by xsdgen.test. DO NOT EDIT.

package ws

import "encoding/xml"

// May be one of A, B, C
type EnumType string

const (
	EnumType_A EnumType = "A"
	EnumType_B EnumType = "B"
	EnumType_C EnumType = "C"
)

type Type1 struct {
	Elem    []EnumType `xml:"Elem,omitempty"`
}
